website/blog/2025/05-19-enhanced-node-management-introducing-in-place-updates-in-gardener.md

```diff
@@ -0,0 +1,84 @@
+*   **`AutoInPlaceUpdate`**: With this strategy, compatible updates are applied directly to the existing nodes. The MachineControllerManager (MCM) automatically selects nodes, cordons and drains them, and then signals the Gardener Node Agent (GNA) to perform the update. Once GNA confirms success, MCM uncordons the node.
+*   **`ManualInPlaceUpdate`**: This strategy also applies updates directly to existing nodes but gives operators fine-grained control. After an update is specified, MCM marks all nodes in the pool as candidates. Operators must then manually label individual nodes to select them for the in-place update process, which then proceeds similarly to the `AutoInPlaceUpdate` strategy.
+
+The `AutoInPlaceUpdate` and `ManualInPlaceUpdate` strategies are available when the `InPlaceNodeUpdates` feature gate is enabled in Gardener.
+
+### What Can Be Updated In-Place?
+
+In-place updates are designed to handle a variety of common operational tasks more efficiently:
+
+*   **Machine Image Updates**: Newer versions of a machine image can be rolled out by executing an update command directly on the node, provided the image and cloud profile are configured to support this.
+*   **Kubernetes Patch Version Updates**: Minor patches to the Kubernetes version of worker nodes can be applied in-place.
+*   **Kubelet Configuration Changes**: Modifications to the Kubelet configuration can be applied directly.
+*   **Credentials Rotation**: Critical for security, rotation of Certificate Authorities (CAs) and ServiceAccount signing keys can now be performed on existing nodes without replacement.
+
+However, some changes still necessitate a rolling update (node replacement):
+*   Changing the machine image name (e.g., switching from Ubuntu to Garden Linux).
+*   Modifying the machine type.
+*   Altering volume types or sizes.
+*   Changing the Container Runtime Interface (CRI) name (e.g., from Docker to containerd).
+*   Enabling or disabling node-local DNS.
+
+### Key API and Component Adaptations
+
+Several Gardener components and APIs have been enhanced to support in-place updates:
+
+*   **CloudProfile**: The `CloudProfile` API now allows specifying `inPlaceUpdates` configuration within `machineImage.versions`. This includes a boolean `supported` field to indicate if a version supports in-place updates and an optional `minVersionForUpdate` string to define the minimum OS version from which an in-place update to the current version is permissible.
+*   **Shoot Specification**: As mentioned, the `Shoot.spec.provider.workers[].updateStrategy` field allows selection of the desired update strategy. Additionally, `Shoot.spec.provider.workers[].machineControllerManagerSettings` now includes `machineInPlaceUpdateTimeout` and `disableHealthTimeout` (which defaults to `true` for in-place strategies to prevent premature machine deletion during lengthy updates). For `ManualInPlaceUpdate`, `maxSurge` defaults to `0` and `maxUnavailable` to `1`.
```

Comment:
```suggestion
*   **Shoot Specification**: As mentioned, the `spec.provider.workers[].updateStrategy` field allows selection of the desired update strategy. Additionally, `spec.provider.workers[].machineControllerManagerSettings` now includes `machineInPlaceUpdateTimeout` and `disableHealthTimeout` (which defaults to `true` for in-place strategies to prevent premature machine deletion during lengthy updates). For `ManualInPlaceUpdate`, `maxSurge` defaults to `0` and `maxUnavailable` to `1`.
```